src/candid_rpc.rs

```diff
@@ -28,29 +28,17 @@ struct CanisterTransport;
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 impl RpcTransport for CanisterTransport {
-    fn resolve_api(
-        service: &cketh_common::eth_rpc_client::providers::RpcService,
-    ) -> Result<cketh_common::eth_rpc_client::providers::RpcApi, cketh_common::eth_rpc::ProviderError>
-    {
-        use crate::candid_rpc::cketh_conversion::{
-            from_rpc_service, into_provider_error, into_rpc_api,
-        };
-        Ok(into_rpc_api(
-            resolve_rpc_service(from_rpc_service(service.clone()))
-                .map_err(into_provider_error)?
-                .api(),
-        ))
+    fn resolve_api(service: &RpcService) -> Result<RpcApi, ProviderError> {
+        Ok(resolve_rpc_service(service.clone())?.api())
```

Comment:
Why is `clone()` needed here? Would it be possible to use a reference to `RpcApi` instead?

Reply:
1. This is the same behaviour as now, so I would not change anything in that PR
2. The trait `RpcTransport` will be removed in #278 
3. clone is necessary because `resolve_rpc_service` takes ownership of `service` but `resolve_api` only has a reference. If ownership is really necessary, then that something we should propagate up to the caller, but I would so this in another PR.